lib/Dialect/FIRRTL/Transforms/Dedup.cpp

```diff
@@ -265,9 +265,9 @@ struct Equivalence {
                       Operation *a, Type aType, Operation *b, Type bType) {
     if (aType == bType)
       return success();
-    if (aType.isa<BundleType>() && bType.isa<BundleType>())
-      return check(diag, message, a, aType.cast<BundleType>(), b,
-                   bType.cast<BundleType>());
+    if (type_isa<BundleType>(aType) && type_isa<BundleType>(bType))
```

Comment:
If possible please prefer dyn_cast over isa + cast, to avoid doing the conversion logic multiple times.  Here, there's short-circuit behavior that may be awkward to preserve so only if you find a simple/neat way to do it.